### ISSUE TITLE
Questions about BatchNorm layer in ResNet in average stage.

### ISSUE THREAD

**Issue by Codefmeister** [TsingZ0/PFL-Non-IID#38]
Hello, thanks for your code.
In my experiment, I found that the set_parameter function in ClientBase.py was implemented by that:
```
        for new_param, old_param in zip(model.parameters(), self.model.parameters()):
            old_param.data = new_param.data.clone()
```
However, I found that this will not transfer the server's running_mean and running_var in batchnorm layer to the corresponding batchnorm layer. So I modified this to:
```
state_dict = model.state_dict()
self.model.load_state_dict(state_dict)
```
I want to know is this wrong? How do we deal with the batchnorm layer in federated learning?  Should we transfer the batchnorm running_mean, running_var to the client? If not, the zero value of this two parameter seems to be harmful to the client model evaluation.

**Comment by KarhouTam** [comment]
Hi.
About transfering running stats of BN layers, maybe you can use `model.state_dict(keep_vars=True)` instead of `model.parameters()`.
`keep_vars=True` makes `model.state_dict()` include all parameters. And parameters of trainable layers like conv still preserve their grad info and `requires_grad=True`.

**Comment by Codefmeister** [comment]
Thanks for your answer. 
I have one more question which seems maybe silly.
Rather than how to transfer running stats, I'd like to know whether should we transfer running stats, especially BN layer to client?
It's really confused me a lot as I am a newbie in Federated Learning.

**Comment by KarhouTam** [comment]
When I do FL experiments, I would always save BN running stats of each client locally, and communicate BN's weight and bias with server at each training round.

**Comment by Codefmeister** [comment]
Thanks for your answer. It helps me a lot. Maybe I should do several exps to validate the assumption.
